### PR TITLE
Read from Note#user association

### PR DESCRIPTION
**Pull request by steventux**
## Do not merge

Depends on https://github.com/DFE-Digital/apply-for-teacher-training/pull/6356

## Context

As part of associating different user types with notes (as authors) we need to read from the new polymorphic `user` association.
This can only be safely done once we're writing to this field and have backfillled any existing notes.

## Changes proposed in this pull request

- Amend notes views to read from `user` field.
- Amend `ApplciationTimelineComponent` to read from the `user` field.

## Guidance to review

Based on https://github.com/DFE-Digital/apply-for-teacher-training/pull/6355

## Link to Trello card

https://trello.com/c/D9WUpK0b/4707-read-from-notesuser-association

## Things to check

- [x] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

File: app/components/provider_interface/application_timeline_component.rb [hunk 1]
```diff
@@ -53,7 +53,7 @@ def note_events
         application_choice.notes.order('created_at').map do |note|
           Event.new(
             'Note added',
-            provider_name(note.provider_user),
+            note.provider_user.full_name,
```
Comment (cwrw) [inline app/components/provider_interface/application_timeline_component.rb:56]: should we switch this up to `user`?

Comment (steventux) [reply app/components/provider_interface/application_timeline_component.rb]: @cwrw updated https://github.com/DFE-Digital/apply-for-teacher-training/pull/6360/commits/4de3f9b228b6821f7c4515ed4b0b0a4c203933a6

**Review by cwrw — APPROVED** [review]
Looks good, could you update the notes export as well? otherwise 👍

**Comment by steventux** [comment]
@cwrw there was a bit of added complexity updating NotesExport as it was very provider user focused. Might be worth taking another look.

**Review by cwrw — APPROVED** [review]
Thanks @steventux 👍 Something to think about is if we want this export to cover the new flavour of notes but this is out of scope for this ticket, and we need to discuss with the BAs before hand later

**Comment by steventux** [comment]
@cwrw :+1:  it will definitely need to be updated to reflect how VendorApiUsers associate with their respective providers.

**Pull request merged**